### ISSUE TITLE
cos_minpoly is too slow

### ISSUE THREAD

**Issue by videlec** [flatsurf/sage-flatsurf#45]
The route via QQbar to obtain the minimal polynomial of 2 cos(pi/n) is not viable.
```
sage: from flatsurf.geometry.subfield import cos_minpoly
sage: cos_minpoly(148)   # does not terminate
```

**Comment by videlec** [comment]
Not so pleasant answer
```
x^72 - 72*x^70 + 2483*x^68 - 54604*x^66 + 860081*x^64 - 10332736*x^62 + 98445520*x^60 - 763615936*x^58 + 4912530808*x^56 - 26567853632*x^54 + 122005197367*x^52 - 479298795820*x^50 + 1619649684367*x^48 - 4726281748816*x^46 + 11940711503788*x^44 - 26157084622864*x^42 + 49703995018057*x^40 - 81887447386216*x^38 + 116801377428793*x^36 - 143899865922244*x^34 + 152619992398647*x^32 - 138741084924896*x^30 + 107508408091128*x^28 - 70526067712544*x^26 + 38839881428636*x^24 - 17773144044704*x^22 + 6673266886466*x^20 - 2024178913320*x^18 + 486479767130*x^16 - 90381694624*x^14 + 12571379640*x^12 - 1254035872*x^10 + 84470637*x^8 - 3511656*x^6 + 77691*x^4 - 684*x^2 + 1
```

**Comment by videlec** [comment]
fixed via #46
